### PR TITLE
Write build-env to file

### LINKED ISSUE #5525: 🐛 BUG: Unable to deploy Pages when using a `worker.toml` with Hyperdrive binding
### Which Cloudflare product(s) does this pertain to?

Pages, Other

### What version(s) of the tool(s) are you using?

Latest everything

### What version of Node are you using?

18.18.2

### What operating system and version are you using?

macOS 14.4.1

### Observed behavior
I am using the Cloudflare Pages Github application to automatically deploy my app. Today, I added the new `wrangler.toml` file since support was just added to Pages. I have a Hyperdrive binding in it. When I push my code to automatically deploy, I get this opaque error message:
```
20:47:19.442	wrangler.toml config file detected
20:47:20.465	Failed: an internal error occurred. If this continues, contact support: https://cfl.re/3WgEyrH
```

### Expected behavior
Because I have seen this issue in development, I assume that the problem is due to this error:
```
Error: When developing locally, you should use a local Postgres connection string to emulate Hyperdrive functionality. Please setup Postgres locally and set the value of the 'WRANGLER_HYPERDRIVE_LOCAL_CONNECTION_STRING_HYPERDRIVE' variable or "HYPERDRIVE"'s "localConnectionString" to the Postgres connection string.
```

When the Cloudflare Pages application builds my Pages site (built with SvelteKit) that has a `wrangler.toml` with a Hyperdrive binding, it needs a Hyperdrive local connection string. How can I provide a local connection string to the Cloudflare Pages application while building? Setting one in the Pages console environment variables and secrets doesn't work.

### Steps to reproduce
1. Create a new SvelteKit project with Cloudflare.
2. Add automatic deployments with the Cloudflare Pages Github Application.
3. Add a `wrangler.toml` file with a Hyperdrive binding.
4. Push your code.
5. Observe that the deployment will fail with a vague error message.

Note that deploying directly using `npx wrangler pages deploy` works because it builds my site locally, which uses my local environment variables that have local connection strings for Hyperdrive. However, I would prefer not to use this method and have my deployments managed by Github.

### Please provide a link to a minimal reproduction

_No response_

### Please provide any relevant error logs

_No response_

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

This ensures that `wrangler pages functions build-env` outputs to a file, rather than stdout.

Fixes #5525 

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: internal only

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]

**Review by petebacondarwin — APPROVED** [review]

File: packages/wrangler/src/pages/build-env.ts [hunk 1]
```diff
@@ -40,9 +41,10 @@ export const Handler = async (args: PagesBuildEnvArgs) => {
 			EXIT_CODE_NO_CONFIG_FOUND
 		);
 	}
+	logger.log("Reading build configuration from your wrangler.toml file...");
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/pages/build-env.ts:44]: will this show up in the CI logs?

File: packages/wrangler/src/pages/build-env.ts [hunk 2]
```diff
@@ -56,14 +58,26 @@ export const Handler = async (args: PagesBuildEnvArgs) => {
...
+	const buildConfiguration = {
+		vars: textVars,
+		pages_build_output_dir: path.relative(
+			args.projectDir,
+			config.pages_build_output_dir
+		),
+	};
+
+	writeFileSync(args.outfile, JSON.stringify(buildConfiguration));
+	logger.debug(`Build configuration written to ${args.outfile}`);
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/pages/build-env.ts:70]: what abt these debug logs?

Comment (penalosa) [reply packages/wrangler/src/pages/build-env.ts:70]: These won't show up in CI

File: packages/wrangler/src/pages/build-env.ts (hunk 1 above)
Comment (penalosa) [reply packages/wrangler/src/pages/build-env.ts:44]: These will, if we print the stdout of this commend. They won't show up by default

Comment (CarmenPopoviciu) [reply packages/wrangler/src/pages/build-env.ts:44]: yes we do

File: packages/wrangler/src/pages/build-env.ts (hunk 2 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/pages/build-env.ts:70]: kk

File: packages/wrangler/src/pages/build-env.ts (hunk 1 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/pages/build-env.ts:44]: I think `build configuration` might be confusing here. perhaps `environment configuration` might be better

File: packages/wrangler/src/pages/build-env.ts [hunk 3]
```diff
@@ -46,13 +58,26 @@ export const Handler = async (args: PagesBuildEnvArgs) => {
...
 
+	const buildConfiguration = {
+		vars: textVars,
+		pages_build_output_dir: path.relative(
+			args.projectDir,
+			config.pages_build_output_dir
+		),
+	};
+
+	writeFileSync(args.outfile, JSON.stringify(buildConfiguration));
```
Comment (CarmenPopoviciu) [inline packages/wrangler/src/pages/build-env.ts:69]: same comment here wrt `build configuration` wording

File: packages/wrangler/src/pages/build-env.ts (hunk 1 above)
Comment (penalosa) [reply packages/wrangler/src/pages/build-env.ts:44]: It includes the build output directory as well, which is why I called it build configuration

**Pull request merged**